pydm/exception.py

```diff
@@ -0,0 +1,82 @@
+import io
+import sys
+import traceback
+from collections import namedtuple
+from qtpy import QtWidgets
+
+
+class QtExceptionHandler:
+    """Utilitary class to handle exception hook and to show exception to
+    operators.
+
+    excepthook is based on https://fman.io/blog/pyqt-excepthook/
+    raise_to_operator is based on https://github.com/pcdshub/typhos/blob/0837405e8d8b73227ac34eb69d2390eed85f3c65/typhos/utils.py#L318
+    """
+    _use_dialog = True
+    _old_excepthook = None
+
+    fake_tb = namedtuple(
+        'fake_tb', ('tb_frame', 'tb_lasti', 'tb_lineno', 'tb_next')
+    )
+
+    @staticmethod
+    def excepthook(exc_type, exc_value, exc_tb):
+        enriched_tb = QtExceptionHandler._add_missing_frames(
+            exc_tb) if exc_tb else exc_tb
+
+        # Note: sys.__excepthook__(...) would not work here.
+        # We need to use print_exception(...):
+        traceback.print_exception(exc_type, exc_value, enriched_tb)
+        if QtExceptionHandler._use_dialog:
+            QtExceptionHandler.raise_to_operator(exc_value)
+
+    @staticmethod
+    def _add_missing_frames(tb):
+        result = QtExceptionHandler.fake_tb(tb.tb_frame, tb.tb_lasti,
```

Comment:
Explanation for why this is necessary?

Reply:
More details here: https://fman.io/blog/pyqt-excepthook/

Reply:
Yeah, there probably should be a couple line comment describing it since it's by no means clear here, and that blog post may not exist forever!

Reply:
Addressed via https://github.com/slaclab/pydm/pull/599/commits/1055892aab1ea1661ba4f8dda43ea77ef7e5bf7c